src/hotspot/share/utilities/rbTree.hpp

```diff
@@ -451,8 +452,55 @@ class RBTree : public AbstractRBTree<K, RBNode<K, V>, COMPARATOR> {
 
 public:
   RBTree() : BaseType(), _allocator() {}
+  NONCOPYABLE(RBTree);
   ~RBTree() { remove_all(); }
 
+  bool copy_into(RBTree& other) const {
+    assert(std::is_copy_constructible<K>::value, "Key type must be copy-constructible when copying a RBTree");
+    assert(std::is_copy_constructible<V>::value, "Value type must be copy-constructible when copying a RBTree");
+    enum Dir { Left, Right };
+    struct node_pair { const RBNode<K, V>* current; RBNode<K, V>* other_parent; Dir d; };
+    struct stack {
+      node_pair s[64];
```

Comment:
Could you clarify the choice of 64 as an upper bound? It's not immediately obvious to me if we guarantee (or if we don't care about) the maximum visitation size in the code below.

Reply:
It's an arbitrary depth (it's used in other parts of the RBTree code as well). The maximum length of the stack represents the maximum depth of the tree that we support. A perfectly balanced binary tree of depth 64 has 2**64 - 1 nodes, so in practice this will never be reached.